web-app/src/components/common/HolidayList/index.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { PropTypes as PT } from 'prop-types';
+import container from './container';
+import { HolidayTable, StatusDot } from './styled';
+import { statusText } from '../../../utilities/holidayStatus';
+
+const buildTableHeaders = (columns) => {
+  return columns.reduce((acc, column) => {
+    return acc.concat(<th key={column}>{column}</th>);
+  }, []);
+};
+
+const buildTableValues = (columns, holiday) => {
+  const { forename, surname } = holiday.employee;
+  const { holidayId, date, dateCreated, holidayStatusId } = holiday;
+
+  const values = {
+    status: <td key={`${holidayId}status`}><StatusDot status={holidayStatusId} />{statusText[holidayStatusId]}</td>,
+    employee: <td key={`${holidayId}employee`}>{`${forename} ${surname}`}</td>,
+    date: <td key={`${holidayId}date}`}>{`${date[2]}/${date[1]}/${date[0]}`}</td>,
+    created: <td key={`${holidayId}created`}>{`${dateCreated[2]}/${dateCreated[1]}/${dateCreated[0]}`}</td>,
+  };
+
+  return columns.reduce((acc, column) => {
+    return acc.concat(values[column]);
+  }, []);
+};
+
+const buildTableActions = (props, holiday) => {
+  const {actions, approveHoliday, rejectHoliday } = props;
+  const { holidayId } = holiday;
+
+  const values = {
+    approve: <button key={`${holidayId}approve`} onClick={() => approveHoliday(holiday)}>Approve</button>,
+    reject: <button key={`${holidayId}reject`} onClick={() => rejectHoliday(holiday)}>Reject</button>,
```

Comment:
Great job on this. I think maybe because the direction this file is going is reusability, maybe these constants could be saved in a separate file?

Reply:
Yep. Barry also mentioned this. I'm going to rework it a little more.